### PR TITLE
PXP-5527 GET /api/v1/objects/{{GUID or ALIAS}}

### PR DESCRIPTION
**Pull request by paulineribeyre**
https://ctds-planx.atlassian.net/browse/PXP-5527

### New Features
- Add GET /api/v1/objects/{{GUID or ALIAS}} endpoint

### CODE REVIEW

File: src/mds/objects.py [hunk 1]
```diff
@@ -119,6 +120,44 @@ async def create_object(
...
+    except httpx.HTTPError as err:
+        logger.debug(err)
+        if err.response and err.response.status_code == 404:
+            logger.debug(
+                f"Could not find GUID or alias {guid} in indexd, querying the metadata database directly"
+            )
+        else:
+            msg = f"Unable to query indexd for GUID or alias {guid}"
+            logger.error(f"{msg}\nException:\n{err}", exc_info=True)
+            # TODO: should we just fall back on MDS DB instead of 500?
```
Comment (paulineribeyre) [inline src/mds/objects.py:155]: @Avantol13 any opinion on this?

Comment (Avantol13) [reply src/mds/objects.py:155]: caveat: didn't look at surrounding code, but I think per the design doc I would expect that we should fall back. The output should include "record" and "metadata", either of which could be empty and that would be valid (200). If the GUID doesn't exist in either indexd or mds then I would expect a 404

File: src/mds/objects.py [hunk 2]
```diff
@@ -119,6 +122,68 @@ async def create_object(
     return JSONResponse(response, HTTP_201_CREATED)
 
 
+@mod.get("/objects/{guid:path}")
+async def get_object(
+    guid, request: Request,
+):
+    """
+    Get the metadata associated with the provided indexd GUID or alias.
+    If the GUID or alias does not exist in indexd, query the metadata
```
Comment (Avantol13) [inline src/mds/objects.py:131]: i would reword this in a way that says it queries both either way. not necessarilly only query mds db if it doesn't exist in indexd

File: src/mds/objects.py [hunk 3]
```diff
@@ -119,6 +122,68 @@ async def create_object(
     return JSONResponse(response, HTTP_201_CREATED)
 
 
+@mod.get("/objects/{guid:path}")
+async def get_object(
+    guid, request: Request,
```
Comment (Avantol13) [inline src/mds/objects.py:127]: guid could use a type hint and we could type hint the return value as well with the `->` syntax

File: tests/test_objects.py [hunk 4]
```diff
@@ -344,3 +344,85 @@ def test_external_api_aliases_failure(client, create_aliases_failure_patcher, da
...
+        resp = client.get(get_object_url)
+        assert indexd_mocked_request.called
+        assert resp.status_code == 200, resp.text
+        assert resp.json() == {"record": indexd_data, "metadata": mds_data}
+    finally:
+        client.delete("/metadata/" + indexd_did)
+
+
+@respx.mock
+def test_get_object_not_in_indexd(client):
```
Comment (Avantol13) [inline tests/test_objects.py:388]: also possible to have an object in indexd but NOT in metadata db, we should test that too

Comment (paulineribeyre) [reply tests/test_objects.py:388]: tested here https://github.com/uc-cdis/metadata-service/blob/b52ccf724feae18117717016d10111984fd7929d/tests/test_objects.py#L366

Comment (Avantol13) [reply tests/test_objects.py:388]: :+1: ah, missed it, thanks

File: run.py [hunk 5]
```diff
@@ -0,0 +1,27 @@
+"""
+Usage:
+- Run app: python run.py
```
Comment (Avantol13) [inline run.py:3]: this is sweet, any way we could automate this in some way? maybe a github action that makes a commit? It's a little annoying/risky for automated commits but I feel like it makes sense for docs, idk if @fantix has an opinion
